# HG changeset patch
# User Rik <rik@octave.org>
# Date 1530556776 25200
#      Mon Jul 02 11:39:36 2018 -0700
# Node ID 6c2d3e9da74270ef03d78d263ee3b7123a248ac4
# Parent  a27ae22eb41d524ac1f43d3a40d9f37a885f64a2
# Parent  60e89abe0e611ab50280e9c4dab2c53e4a7f0bfa
maint: merge stable to default.

diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -411,17 +411,17 @@ function [local_packages, global_package
           if (! all (succ))
             i = find (! succ, 1);
             error ("pkg: could not download file %s from url %s",
                    local_files{i}, urls{i});
           endif
         else
           ## If files do not exist, maybe they are not local files.
           ## Try to download them.
-          external_files_mask = ! cellfun (@exist, files, {"file"});
+          external_files_mask = cellfun (@(x) isempty (glob (x)), files);
           if (any (external_files_mask))
             [success, msg] = mkdir (tmp_dir);
             if (success != 1)
               error ("pkg: failed to create temporary directory: %s", msg);
             endif
 
             for file_idx = find (external_files_mask)
 
